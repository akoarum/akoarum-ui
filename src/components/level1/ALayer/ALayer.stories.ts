import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import { withKnobs } from '@storybook/addon-knobs'
import { withInfo } from 'storybook-addon-vue-info'
import ALayer from './ALayer.vue'

storiesOf('level1/ALayer', module)
  .addDecorator(withInfo)
  .addDecorator(withKnobs)
  .add('Normal', () => ({
    components: { ALayer },
    template: `
      <ALayer :style="{ width: '150px', height: '150px' }" @click="click" />
    `,
    methods: {
      click: action('click')
    }
  }), { info: {} })
