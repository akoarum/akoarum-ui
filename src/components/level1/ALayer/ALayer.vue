<template>
  <div class="aLayer" @click="handleClick" />
</template>

<script lang="ts">
import { defineComponent, SetupContext } from '@vue/composition-api'

export default defineComponent({
  setup(_, context: SetupContext) {
    const handleClick = () => {
      context.emit('click')
    }

    return { handleClick }
  }
})
</script>

<style lang="scss" scoped>
.aLayer {
  background: rgba(0, 0, 0, 0.65);
}
</style>
