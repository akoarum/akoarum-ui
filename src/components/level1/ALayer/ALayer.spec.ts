import { shallowMount, createLocalVue } from '@vue/test-utils'
import VueCompositionApi from '@vue/composition-api'
import ALayer from './ALayer.vue'

describe('ALayer', () => {
  const localVue = createLocalVue()
  let wrapper

  localVue.use(VueCompositionApi)

  beforeEach(() => {
    wrapper = shallowMount(ALayer, {
      propsData: {},
      localVue
    })
  })

  describe('initialized', () => {
    it('mountable', () => {
      expect(wrapper.isVueInstance()).toBe(true)
    })
    it('snapshot', () => {
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('event', () => {
    it('要素をクリックするとclickイベントを通知する', () => {
      wrapper.trigger('click')
      expect(wrapper.emitted().click).toBeTruthy()
    })
  })
})
