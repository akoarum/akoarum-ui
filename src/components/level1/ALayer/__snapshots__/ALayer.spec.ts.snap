// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ALayer initialized snapshot 1`] = `
<div
  class="aLayer"
/>
`;
